src/use-utility-connect.js

```diff
@@ -16,32 +19,59 @@ const getConfigError = errors => {
 };
 
 export const useUtilityConnect = () => {
-  const [error, setError] = useState();
-  const [loading, setLoading] = useState(false);
+  const [utilityConnectError, setUtilityConnectError] = useState();
+  const [widgetOpening, setWidgetOpening] = useState(false);
   const [factory, setFactory] = useState();
 
   const [openOnScriptLoad, setOpenOnScriptLoad] = useState(false);
   const [savedOpenParams, setSavedOpenParams] = useState();
+  const pollForFactoryTimeout = useRef(null);
+
+  const pollingCount = useRef(0);
```

Comment:
This is me asking fundamental react questions -- what would happen if we didn't use `useRef` here and just instead set the values directly. 

```javascript
const pollForFactoryTimeout = null
const pollingCount = 0
```

Guessing the values would not persist correctly?

Reply:
Yeah. They wouldn't be propagated to the function, which would have stale state.